manim/scene/scene_file_writer.py

```diff
@@ -732,21 +751,17 @@ def combine_to_movie(self):
             # but tries to call ffmpeg via its CLI -- which we want
             # to avoid. This is why we need to do the conversion
             # manually.
-            if config.format == "webm":
-                with (
-                    av.open(sound_file_path) as wav_audio,
-                    av.open(sound_file_path.with_suffix(".ogg"), "w") as opus_audio,
-                ):
-                    wav_audio_stream = wav_audio.streams.audio[0]
-                    opus_audio_stream = opus_audio.add_stream("libvorbis")
-                    for frame in wav_audio.decode(wav_audio_stream):
-                        for packet in opus_audio_stream.encode(frame):
-                            opus_audio.mux(packet)
-
-                    for packet in opus_audio_stream.encode():
-                        opus_audio.mux(packet)
-
-                sound_file_path = sound_file_path.with_suffix(".ogg")
+            out_suffix = movie_file_path.suffix.lower()
+            if config.format == "webm" or out_suffix == ".webm":
+                ogg_sound_file_path = sound_file_path.with_suffix(".ogg")
+                convert_audio(sound_file_path, ogg_sound_file_path, "libvorbis")
+                sound_file_path = ogg_sound_file_path
+            elif config.format == "mp4" or out_suffix == ".mp4":
```

Comment:
In the case where rendering with `--format=mp4 --transparent`, we internally construct a `.mov`, but keep the `config.format` value to be `.mp4`. This would cause manim stepping inside this conditional, which is probably not intended (but still works, apparently).

Should we (still) just compare against `config.format`?

```suggestion
            elif config.format == "mp4":
```

Reply:
Same thing here, `config.format` may be `None`.

> we internally construct a .mov

I don't understand this part. Libav is requested to produce a file with a .mp4 suffix. How can it know its format should really be that of a .mov?

Reply:
I've just checked; this is what I meant: https://github.com/ManimCommunity/manim/blob/d6f066c1318a3d131823926b7a5d31c2aaa48ccb/manim/_config/utils.py#L1424-L1438

The source of truth concerning which sort of container is requested can actually be found in `config.movie_file_extension`. I also think that this property can never be `None`, so I'd consider it safe to check against.